### ISSUE TITLE
Investigate how Eternl behaves with UTxO locking feature

### ISSUE THREAD

**Issue by klntsky** [Plutonomicon/cardano-transaction-lib#1282]
`walletWaitForInputs` may never finish, because getUtxos removes the locked utxos from the response, that makes CTL think that it hasn't synchronized yet.

This should be confirmed by running a real test.

**Comment by Luis-omega** [comment]
I can confirm that `Eternl.getUxos` function in fact hides the locked utxos. 

The main problem is this line : 

```
      unless (ownInputUtxos `Map.isSubmap` walletUtxos) do
```

Where `ownIputUtxos` is obtained without using a particular wallet and `walletUtxos` is obtained by `getUxos` from eternl.

**Comment by Luis-omega** [comment]
Results:  If a transaction in the inputs is locked, `walletWaitForInputs` would stop with the expected error defined inside the function after the defined time. 

Procedure to test : 

I used this modified `Pkh2Pkh` example : 
```

module Ctl.Examples.Pkh2Pkh (main, contract, example) where

import Contract.Prelude

import Contract.Address (ownPaymentPubKeysHashes, ownStakePubKeysHashes)
import Contract.Config (ConfigParams, testnetNamiConfig)
import Contract.Log (logInfo')
import Contract.Monad (Contract, launchAff_, liftedE, liftedM, runContract)
import Contract.ScriptLookups as Lookups
import Contract.Transaction (awaitTxConfirmedWithTimeout, balanceTx, signTransaction)
import Contract.TxConstraints as Constraints
import Contract.Value as Value
import Ctl.Examples.Helpers (buildBalanceSignAndSubmitTx) as Helpers
import Data.Array (head)
import Data.BigInt as BigInt
import Effect.Aff (delay)

main :: Effect Unit
main = example testnetNamiConfig

contract :: Contract () Unit
contract = do
  logInfo' "Running Examples.Pkh2Pkh"
  pkh <- liftedM "Failed to get own PKH" $ head <$> ownPaymentPubKeysHashes
  skh <- liftedM "Failed to get own SKH" $ join <<< head <$>
    ownStakePubKeysHashes

  let
    constraints :: Constraints.TxConstraints Void Void
    constraints = Constraints.mustPayToPubKeyAddress pkh skh
      $ Value.lovelaceValueOf
      $ BigInt.fromInt 2_000_000

    lookups :: Lookups.ScriptLookups Void
    lookups = mempty

  unbalancedTx <- liftedE $ Lookups.mkUnbalancedTx lookups constraints
  balancedTx <- liftedE $ balanceTx unbalancedTx
  logInfo' $ "transaction until now"
  logInfo' $ show balancedTx
  wait 30000.0
  balancedSignedTx <- signTransaction balancedTx
  logInfo' $ show balancedSignedTx
  pure unit


example :: ConfigParams () -> Effect Unit
example cfg = launchAff_ do
  runContract cfg contract


wait :: Number -> Contract () Unit
wait n = unless (n ==0.0) do 
  logInfo' $ show n
  liftAff $ delay $ wrap 1000.0
  wait (n-1000.0)
```

Then use the `e2e-browser` to run it and get access by hand to the wallet. When the transaction was ready to be signed I locked the input and wait (I put `logInfo` in `walletWaitFroInputs` to get a countdown). The contract throw the expected error every time I tried it (5 times).

**Comment by klntsky** [comment]
The results of investigation lead to changes implemented in  #1440
